pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -403,7 +396,11 @@ def barrier(self, *args, **kwargs) -> None:
             torch.distributed.barrier()
 
     def broadcast(self, obj: object, src: int = 0) -> object:
-        return self.dist.broadcast(obj)
+        obj = [obj]
+        if self.global_rank != 0:
```

Comment:
```suggestion
        if self.global_rank != src:
```
does this need to be the src instead of hardcoded to rank 0?

Reply:
I think the old assumption is that torch distributed broadcast only works with broadcast from global rank 0. WIth respect the change respect src, i think here should be change to src as well. Good catch